common/beerocks/bcl/unit_tests/wireless_utils_test.cpp

```diff
@@ -13,120 +13,265 @@
+ * create a valid sWifiChannel to use as parameter to get_operating_class_by_channel() function and
+ * the expected operating class to be obtained as result.
+ *
+ * @return vector with pairs of valid WiFi channels and their expected operating class result.
+ */
+std::vector<tGetOperatingClassByChannelParam> get_operating_class_by_channel_valid_parameters()
 {
-    EXPECT_EQ(get_operating_class_by_channel_helper(1, 20), 81);
-    EXPECT_EQ(get_operating_class_by_channel_helper(13, 20), 81);
+    std::vector<tGetOperatingClassByChannelParam> result;
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(14, 20), 82);
+    for (const auto &entry : table_E_4_global_operating_classes_802_11) {
+        for (const auto channel : entry.channels) {
+            beerocks::message::sWifiChannel ch;
+            ch.channel           = get_starting_channel(channel, entry.bandwidth);
+            ch.channel_bandwidth = entry.bandwidth;
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(1, 40), 83);
-    EXPECT_EQ(get_operating_class_by_channel_helper(9, 40), 83);
+            result.emplace_back(std::make_tuple(ch, entry.operating_class));
+        }
+    }
 
-    /* Note: we have no way to distinguish 40- and 40+ */
-    EXPECT_EQ(get_operating_class_by_channel_helper(10, 40), 84);
-    EXPECT_EQ(get_operating_class_by_channel_helper(13, 40), 84);
+    return result;
+}
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(36, 20), 115);
-    EXPECT_EQ(get_operating_class_by_channel_helper(48, 20), 115);
+/**
+ * @brief Gets a list of invalid parameters to get_operating_class_by_channel() function.
+ *
+ * Creates a container of sWifiChannel objects built up with invalid combinations of channel
+ * number and channel bandwidth. The get_operating_class_by_channel() function returns 0 when
+ * called with any of these invalid channels as parameter.
+ *
+ * @return vector of invalid WiFi channels.
+ */
+std::vector<tGetOperatingClassByChannelParam> get_operating_class_by_channel_invalid_parameters()
+{
+    // clang-format off
+    const beerocks::eWiFiBandwidth bandwidths[]{
+        beerocks::BANDWIDTH_UNKNOWN,
+		beerocks::BANDWIDTH_20,
+        beerocks::BANDWIDTH_40,
+		beerocks::BANDWIDTH_80,
+        beerocks::BANDWIDTH_80_80,
+		beerocks::BANDWIDTH_160
+    };
+    // clang-format on
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(36, 40), 116);
-    EXPECT_EQ(get_operating_class_by_channel_helper(44, 40), 116);
+    std::vector<tGetOperatingClassByChannelParam> result;
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(40, 40), 117);
-    EXPECT_EQ(get_operating_class_by_channel_helper(48, 40), 117);
+    /**
+     * Note: Google Test documentation says that using value-parameterized tests to test your code
+     * over various inputs (a.k.a. data-driven testing) is a feature easy to abuse of.
+     * This might well be a good example of such an abuse.
+     */
+    uint8_t channel = 0;
+    do {
```

Comment:
Why do-while instead of for loop?

Reply:
So that the loop executes up until and INCLUDING `channel == UINT8_MAX`, I guess (`for (uint8_t channel = 0; channel <= UINT8_MAX; ++channel)` would loop forever)

Reply:
Ah of course... If we make channel an unsigned instead of a uint8_t, that problem goes away.

Reply:
Exactly!
I'm going to add a comment to clarify it